client/src/main/scala/org/apache/celeborn/client/commit/CommitHandler.scala

```diff
@@ -439,7 +442,7 @@ abstract class CommitHandler(
         replicaIds)
     }
 
-    doParallelCommitFiles(shuffleId, shuffleCommittedInfo, params, commitFilesFailedWorkers)
+    doParallelCommitFiles(shuffleId, shuffleCommittedInfo, params, commitFilesFailedWorkers, true)
```

Comment:
Only the ReducePartitionCommitHandler can set stageEnd to true when call handleFinalCommitFiles method.